test/nimble_strftime_test.exs

```diff
@@ -131,12 +131,20 @@ defmodule NimbleStrftimeTest do
       assert NimbleStrftime.format(%{month: 12}, "%B %b") == "December Dec"
     end
 
+    test "microseconds format ignores padding and width options" do
+      datetime = ~U[2019-08-15 17:07:57.001234Z]
+      assert NimbleStrftime.format(datetime, "%f") == "001234"
+      assert NimbleStrftime.format(datetime, "%f") == NimbleStrftime.format(datetime, "%_20f")
+      assert NimbleStrftime.format(datetime, "%f") == NimbleStrftime.format(datetime, "%020f")
+      assert NimbleStrftime.format(datetime, "%f") == NimbleStrftime.format(datetime, "%-f")
```

Comment:
We need more tests. Especially tests that include:

1. Naive datetimes with different precisions (.1, .123, .123456)
2. Naive datetimes with 0 micrsoeconds in different precisions (.0, .00, .000000)
3. Naive datetimes with no precision (which we may have to round to `"0"`